pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -30,6 +30,17 @@ type Sync struct {
 	ProviderArgs sync.ProviderArgs
 	client       client.Client
 	URI          string
+	ready        bool
+}
+
+func (k *Sync) Init(ctx context.Context) error {
+	// noop
+	return nil
+}
+
+func (k *Sync) IsReady() bool {
+	// we cannot reliably check external HTTP(s) sources
```

Comment:
I think this comment is an error (it seems like it might have been copied from the http impl.